### ISSUE TITLE
memoryview not accepted in str constructor

### ISSUE THREAD

**Issue by anuraaga** [astral-sh/ty#3243]
### Summary

It seems `memoryview` is rejected for `str` constructor with encoding param. AFAICT, this is allowed by Python.

```python
buf = memoryview(b"hello world")
s = str(buf, "utf-8")
```

```
error[[invalid-argument-type](https://ty.dev/rules#invalid-argument-type)]: Argument to class `str` is incorrect
 --> str_memoryview.py:5:9
  |
3 | buf = memoryview(b"hello world")
4 |
5 | s = str(buf, "utf-8")
  |         ^^^ Expected `bytes | bytearray`, found `memoryview[int]`
6 | print(s)
  |
info: rule `invalid-argument-type` is enabled by default
```

https://docs.python.org/3/library/stdtypes.html#textseq

> If at least one of encoding or errors is given, object should be a [bytes-like object](https://docs.python.org/3/glossary.html#term-bytes-like-object)

https://docs.python.org/3/glossary.html#term-bytes-like-object

> An object that supports the [Buffer Protocol](https://docs.python.org/3/c-api/buffer.html#bufferobjects) and can export a C-[contiguous](https://docs.python.org/3/glossary.html#term-contiguous) buffer. This includes all [bytes](https://docs.python.org/3/library/stdtypes.html#bytes), [bytearray](https://docs.python.org/3/library/stdtypes.html#bytearray), and [array.array](https://docs.python.org/3/library/array.html#array.array) objects, as well as many common [memoryview](https://docs.python.org/3/library/stdtypes.html#memoryview) objects.

I'm not sure what the typing distinction would be for a "common" vs "uncommon" memoryview as described here, but I guess `memoryview[int]` at the least is supposed to work.

### Version

ty 0.0.29 (438a78d68 2026-04-05)

**Comment by carljm** [comment]
Thanks! Yes, this is a bug in ty; it looks like at some point we hardcoded the `str` constructor, but simplified it from the actual signature. We should just remove that hardcoded constructor signature.
